pkg/controller/container-runtime-config/container_runtime_config_controller.go

```diff
@@ -207,9 +234,24 @@ func (ctrl *Controller) Run(workers int, stopCh <-chan struct{}) {
 	defer utilruntime.HandleCrash()
 	defer ctrl.queue.ShutDown()
 	defer ctrl.imgQueue.ShutDown()
+	listerCaches := []cache.InformerSynced{ctrl.mcpListerSynced, ctrl.mccrListerSynced, ctrl.ccListerSynced,
+		ctrl.imgListerSynced, ctrl.icspListerSynced, ctrl.idmsListerSynced, ctrl.itmsListerSynced, ctrl.clusterVersionListerSynced}
+
+	enabled, err := ctrl.sigstoreAPIEnabled()
+	if enabled {
+		err = ctrl.addImagePolicyObservers()
+		if err != nil {
+			klog.Infof("skipping error adding image policy observers: %v", err)
```

Comment:
If `New` finds the gate enabled, `err` will be “already added policy observers”.
- Isn’t this unnecessarily noisy?
- AFAICT we would not run `Start` and update `listerCaches` in that case.

Consider moving the `addedPolicyObservers` field handling into the callers (or, see elsewhere, removing the code in `New`).